### PR TITLE
DSEGOG-135 Channel Search

### PR DESCRIPTION
**Pull request by louise-davies**
## Description
Add the ability to search for channels using a MUI autocomplete, which upon selecting a channel will show that channel in the metadata panel & show the correct place in the channel tree. Channels are searchable by both friendly names and system names.

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes DSEGOG-135

### CODE REVIEW

File: src/channels/channelsDialogue.component.test.tsx [hunk 1]
```diff
@@ -183,4 +183,18 @@ describe('Channels Dialogue', () => {
...
+
+  it('selects a channel in the metadata panel & channel tree when search box is used', async () => {
+    createView();
+
+    const search = screen.getByLabelText('Search data channels');
+
+    await user.type(search, 'shot{arrowdown}{enter}');
+
+    // shot number should be visible in tree view
+    screen.getByRole('checkbox', { name: 'Shot Number' });
```
Comment (kennethnym) [inline src/channels/channelsDialogue.component.test.tsx]: Should this be wrapped in an `expect` assertion?
```js
expect(screen.getByRole('checkbox', { name: 'Shot Number' })).toBeInTheDocument();
```

File: src/channels/channelsDialogue.component.test.tsx [hunk 2]
```diff
@@ -183,4 +183,18 @@ describe('Channels Dialogue', () => {
...
+
+    // shot number should be selected in the metadata panel
+    screen.getByText('System name: shotnum');
```
Comment (kennethnym) [inline src/channels/channelsDialogue.component.test.tsx]: Same as above.

**Review by kennethnym — CHANGES_REQUESTED** [review]
Code looks good and works well, just one minor comment :)

File: src/channels/channelsDialogue.component.test.tsx (hunk 1 above)
Comment (louise-davies) [reply src/channels/channelsDialogue.component.test.tsx]: Technically it doesn't have to be, as `getByRole` would error if it wasn't in the document, but I can wrap in `expect` to make it more explicit

Comment (kennethnym) [reply src/channels/channelsDialogue.component.test.tsx]: Yeah, I think wrapping it in an expect makes the intention clearer, especially for devs who might not be aware that `getBy*` queries throw when no element is found.

**Review by kennethnym — APPROVED** [review]
LGTM!

**Pull request merged**